SmartTutor/KnowledgeComponentModel/KnowledgeComponents/KnowledgeComponent.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+
+namespace SmartTutor.KnowledgeComponentModel.KnowledgeComponents
+{
+    public class KnowledgeComponent
+    {
+        public int Id { get; private set; }
+        
+        public string Name { get; private set; }
+
+        public List<KnowledgeComponent> KnowledgeSubcomponents { get; private set; }
```

Comment:
subcomponents is a sufficient name. Not sure if we need this though, as a field. In general, we want to minimize the amount of associations and (reduce bidirectional to unidirectional where possible) to support our use cases and nothing else.

Reply:
Isn't this unidirectional relationship and removing it would remove relation as well?
Would it be better to store `Subcomponents` or `Parent`, we were discussing and settled on `Subcomponents`, what are your thoughts? @Luburic